src/test/js/shared/idam_helper.js

```diff
@@ -361,43 +363,21 @@ class IdamHelper extends Helper {
             .catch(err => err);
     }
 
-    getEmail(searchEmail) {
-        return (
-            notifyClient
-                .getNotifications("email", null)
-                .then(response => {
-                    console.log("Searching " + response.body.notifications.length + " emails(s) from sending queue");
-                    return this.searchForEmailInResults(response.body.notifications, searchEmail);
-                })
-                .then(emailResponse => {
-                    if (emailResponse) {
-                        return emailResponse;
-                    } else {
-                        return (
-                            notifyClient.getNotifications("email", "failed")
-                                .then(failedResponse => {
-                                    console.log("Searching " + failedResponse.body.notifications.length + " emails(s) from failure queues");
-                                    return this.searchForEmailInResults(failedResponse.body.notifications, searchEmail);
-                                })
-                                .then(failedEmailResponse => {
-                                    if (failedEmailResponse) {
-                                        return failedEmailResponse;
-                                    } else {
-                                        throw new Error('No emails found for ' + searchEmail);
-                                    }
-                                })
-                        );
-                    }
-                })
-        );
-    }
-
-    async extractUrl(searchEmail) {
-        const emailResponse = await this.getEmail(searchEmail);
-        return this.extractUrlFromBody(emailResponse);
-    }
-
-    searchForEmailInResults(notifications, searchEmail) {
+    async getEmailFromNotify(searchEmail) {
+        let notificationsResponse = await notifyClient.getNotifications("email", null);
+        let emailResponse = this.searchForEmailInNotifyResults(notificationsResponse.body.notifications, searchEmail);
+        let i = 0;
+        while (i < maxNotifyPages && !emailResponse && notificationsResponse.body.links.next) {
+            console.log("Searching notify emails, next page " + notificationsResponse.body.links.next);
+            let olderThanId = notificationsResponse.body.links.next.match('[0-9a-fA-F]{8}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{12}');
```

Comment:
the regex makes me a little nervous - It makes us very reliant on their id structure, which they could theoretically change. I think this is coming from an URI so could we not base it on the name of the attribute that we want?